listener.go

```diff
@@ -345,8 +345,20 @@ func (l *listener) getCertificate(hello *tls.ClientHelloInfo) (*tls.Certificate,
 			return nil, err
 		}
 	}
-
-	return l.loadCert(newConn)
+	connCert, err := l.loadCert(newConn)
+	if connCert != nil && err == nil && newConn != nil && l.conns != nil {
+		// if we were successfully able to load a cert and are closing connections on cert changes, mark newConn ready
+		// this will allow us to close the connection if a future connection forces the cert to re-load
+		wrapper, ok := newConn.(*closeWrapper)
+		if !ok {
+			logrus.Infof("will not mark non-close wrapper connection from %s to %s as ready", newConn.RemoteAddr(), newConn.LocalAddr())
```

Comment:
This could get noisy, I might downgrade this to Debugf

Reply:
This looks kinda like development logging that could just go away entirely. The connection will not be wrapped by closeWrapper in most non-Rancher projects where we don't ask for the close-on-change behavior.

Reply:
I'll downgrade to debug - I think that's more consistent with project behavior. 

> This looks kinda like development logging that could just go away entirely. The connection will not be wrapped by closeWrapper in most non-Rancher projects where we don't ask for the close-on-change behavior.

@brandond In those cases, I would expect that `l.conns == nil` and this code to never be invoked. I want this debug to be in place because the population of l.conns indicates that you wanted the close-on-change behavior, but we weren't able to provide it because of improper wrapping. See here: https://github.com/rancher/dynamiclistener/blob/401fafb7e6dfec2b55aa0008f51640e8a8bb3fce/listener.go#L71-L76 for what I mean on expecting this block to only be called when you asked for close-on-change behavior.

Reply:
In that case would it be an error? Informational says it's something that a user would want to know about. If it indicates a problem in our code, lets make it an error. If it doesn't matter to the user lets make it a debug or trace.

Reply:
From a strictly technical perspective this could probably be best classified as an error. But I think @cmurphy 's concern still stands in that case - if we are wrong this will flood people's logs.

Reply:
Fair enough. Are you still planning on downgrading this to a debug? LGTM otherwise.

Reply:
Yep - I'll downgrade to debug and push this up. Sorry for the delay.

Reply:
Resolving, this now uses `Debugf`